2d/src/main/java/org/tweetwallfx/twod/TagTweets.java

```diff
@@ -177,25 +154,23 @@ public void stop() {
     private class TweetsCreationTask extends Task<Void> {
 
         private final String searchText;
-        private TwitterStream stream;
-        private final Configuration conf;
-        private final BlockingQueue<TweetInfo> tweets;
+        private TweetStream stream;
+        private final Tweeter tweeter;
+        private final BlockingQueue<Tweet> tweets;
 
-        public TweetsCreationTask(Configuration conf, String searchText, BlockingQueue<TweetInfo> tweets) {
-            this.conf = conf;
+        public TweetsCreationTask(Tweeter tweeter, String searchText, BlockingQueue<Tweet> tweets) {
```

Comment:
I was able to extract the TweetsCreationTask into the TweetSetData storage object keeping track of the underlying data being displayed in whatever view is desired